frontend/src/components/playground/TransformationItem.tsx

```diff
@@ -0,0 +1,149 @@
+
+  const handleEditNameChange = (
+    event: CustomEvent<ButtonProps.ClickDetail>,
+  ) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (isEditing) {
+      if (editName.trim()) {
+        updateTransformation(item.id, editName, item.data.content);
+      } else {
+        setEditName(item.data.name); // Reset to original if empty
+      }
+
+      // Exit edit mode
+      setIsEditing(false);
+    } else {
+      // Enter edit mode
+      setEditName(item.data.name);
+      setIsEditing(true);
+    }
+  };
+
+  return (
+    <BoardItem
+      key={item.id}
+      header={
+        <Header
+          actions={
+            <SpaceBetween direction="horizontal" size="xs">
+              <Button
+                variant="inline-icon"
+                iconName={isEditing ? "check" : "edit"}
+                ariaLabel={
+                  isEditing
+                    ? `Save ${item.data.name}`
+                    : `Edit ${item.data.name}`
+                }
+                onClick={handleEditNameChange}
+              />
+              <Button
+                variant="inline-icon"
+                iconName="script"
+                ariaLabel={`Format code in ${item.data.name}`}
+                onClick={(e) => {
+                  e.preventDefault();
+                  e.stopPropagation();
+                  if (formatCodeRef.current) {
+                    formatCodeRef.current();
+                  }
+                }}
+              />
+              <Button
+                variant="inline-icon"
+                iconName="remove"
+                ariaLabel={`Delete ${item.data.name}`}
+                onClick={(e) => {
+                  e.preventDefault();
+                  e.stopPropagation();
+                  onRemove(item.id);
+                }}
+              />
+            </SpaceBetween>
+          }
+        >
+          <div style={{ display: "flex", alignItems: "center" }}>
+            {isEditing ? (
+              <Input
+                value={editName}
+                onChange={({ detail }) => setEditName(detail.value)}
+                onKeyDown={({ detail }) => {
+                  if (detail.key === "Enter") {
+                    if (editName.trim()) {
+                      updateTransformation(
+                        item.id,
+                        editName,
+                        item.data.content,
+                      );
+                    } else {
+                      setEditName(item.data.name); // Reset to original if empty
+                    }
+                    setIsEditing(false);
+                  } else if (detail.key === "Escape") {
+                    setEditName(item.data.name); // Reset to original
+                    setIsEditing(false);
+                  }
+                }}
+                autoFocus
+              />
+            ) : (
+              <>
+                {item.data.name}
+                <div style={{ marginLeft: "10px" }}>
```

Comment:
Is there a way we can get this formatting using the Cloudscape layout primitives instead of CSS?

Reply:
If we've got to push pixels, I'd rather they were only inside the AceEditorComponent, and everything out side it could cleanly use the cloudscape theming.  What do you think?

Reply:
Yes, probably, I'll play around with it.